dds_web/api/user.py

```diff
@@ -876,34 +876,77 @@ def post(self):
-                f"The user with email '{user_email}' does not have access to the specified project."
-                " Cannot remove non-existent project access."
-            )
+        if unanswered_invite:
+            # If there is a unit_id value, it means the invite was associated to a unit
+            # i.e The invite is for a Unit Personel which shouldn't be removed from individual projects
+            if unanswered_invite.unit_id:
+                raise ddserr.UserDeletionError(
+                    "Cannot remove a Unit Admin / Unit User from individual projects"
+                )
+
+            invite_id = unanswered_invite.id
+
+            # Check if the unanswered invite is associated with the project
+            project_invite_key = models.ProjectInviteKeys.query.filter_by(
+                invite_id=invite_id, project_id=project.id
+            ).one_or_none()
+
+            if project_invite_key:
+                # Remove the association if it exists
+                db.session.delete(project_invite_key)
+
+                # Check if the invite is associated with only one project, if it is -> delete the invite
+                project_invite_key = models.ProjectInviteKeys.query.filter_by(
+                    invite_id=invite_id
+                ).one_or_none()
+
+                if not project_invite_key:
+                    db.session.delete(unanswered_invite)
+            else:
+                # The unanswered invite is not associated with the project
+                raise ddserr.NoSuchUserError(
+                    f"The user / invite with email '{user_email}' does not have access to the specified project. "
+                    "Cannot remove non-existent project access."
+                )
+
+        else:
+            if auth.current_user().username == existing_user.username:
+                raise ddserr.AccessDeniedError(message="You cannot renew your own access.")
+
+            # Search the user in the project, when found delete from the database all references to them
+            user_in_project = False
+            for (
+                user_association
+            ) in project.researchusers:  # TODO Possible optimization -> comprehesion list
+                if user_association.user_id == existing_user.username:
+                    user_in_project = True
+                    db.session.delete(user_association)
+                    project_user_key = models.ProjectUserKeys.query.filter_by(
+                        project_id=project.id, user_id=existing_user.username
+                    ).first()
+                    if project_user_key:
+                        db.session.delete(project_user_key)
+                break
+
+            if not user_in_project:
+                raise ddserr.NoSuchUserError(
+                    f"The user / invite with email '{user_email}' does not have access to the specified project. "
+                    "Cannot remove non-existent project access."
+                )
```

Comment:
Add message here instead
```suggestion
                )
            msg = f"User with email {user_email} no longer associated with {project.public_id}."
```